src/main/java/com/rocketden/main/service/SubmitService.java

```diff
@@ -58,4 +105,58 @@ public SubmissionDto submitSolution(Game game, SubmissionRequest request) {
 
         return GameMapper.submissionToDto(submission);
     }
+
+    // Sends a POST request to the tester service to judge the user submission
+    protected Submission callTesterService(TesterRequest request) {
+        try {
+            HttpPost post = new HttpPost(getTesterUrl());
+
+            StringEntity stringEntity = new StringEntity(gson.toJson(request));
+            post.setEntity(stringEntity);
+            post.setHeader("Content-type", "application/json");
+
+            HttpResponse response = httpClient.execute(post);
+            String jsonResponse = EntityUtils.toString(response.getEntity());
+
+            TesterResponse testerResponse = getResponseFromJson(jsonResponse);
+
+            // TODO: logic to convert TesterResponse into Submission object
+            Submission submission = new Submission();
+            submission.setNumCorrect(request.getProblem().getTestCases().size());
+            submission.setNumTestCases(request.getProblem().getTestCases().size());
+            submission.setPlayerCode(new PlayerCode(request.getCode(), request.getLanguage()));
+
+            return submission;
+        } catch (Exception e) {
+            // If in debug mode (tester is unavailable), return a dummy submission
+            if (getDebugMode()) {
+                return getDummySubmission(request);
+            }
```

Comment:
Why not put this if statement at the start of the method, outside of the try-catch block? Is the testerService still supposed to return something if you are in debug mode?

Reply:
My main thinking behind it was that the dummy submission would be an optional fallback for the actual request. Obviously rn the request is not yet ready so it always fails/falls back - if you think this might be better suited to be put at the start, lmk and I can do that

Reply:
Yeah I think it makes more sense for it to be at the start; if you're testing the actual request, you'd want it to show the error to troubleshoot rather than a dummy submission imo.

Reply:
Sounds good - implemented